src/types.ts

```diff
@@ -0,0 +1,35 @@
+export interface RGB {
+  r: number;
+  g: number;
+  b: number;
+  [key: string]: number;
+}
+
+export interface HSL {
+  h: number;
+  s: number;
+  l: number;
+  [key: string]: number;
+}
+
+export interface HSV {
+  h: number;
+  s: number;
+  v: number;
+  [key: string]: number;
+}
```

Comment:
May I ask what's the purpose of `[key: string]: number;` lines? Seems strange to me because these objects shouldn't have extra keys.

Reply:
It's required for your `equalColorObjects` function. TS can't index an object unless you define how it's indexed.

You'll get the following error, though for HSL & HSV too:

```
Type 'RGB' is not assignable to type 'Record<string, number>'.
Index signature is missing in type 'RGB'.
```

Reply:
Got it. Thanks for your explanation!